### ISSUE TITLE
Example with nextjs doesn't work with React hooks

### ISSUE THREAD

**Issue by Siegrift** [firebase/functions-samples#601]
### How to reproduce these conditions
I have set up an example [repo](https://github.com/Siegrift/nextjs-firebase-example) where you can check the bug.

**Steps to set up and reproduce**
1) `git clone https://github.com/Siegrift/nextjs-firebase-example`
2) `cd src/app`
3) `yarn && yarn dev`
4) navigate to `http://localhost:3000/about`
**You will get:**
Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

### Expected behavior

I don't think that this example is valid anymore. I was trying to figure out how to make nextjs work with firebase, but I couldn't figure it out.... Anyway, the expected behaviour is to make it work with nextjs and newest React (e.g. react hooks).

### Actual behavior
Error in application and I am unable to configure nextjs with firebase.

**Comment by maotora** [comment]
Did you solve this?
I'm experiencing the same issue.

**Comment by Siegrift** [comment]
Unfortunately no, but I didn't needed that much. I use firebase and firestore and I've set it up manually inside the project (https://firebase.google.com/docs/web/setup)

**Comment by maotora** [comment]
So you are hosting your web app on static hosting?
My Nextjs + Antd app got really slow as the size grew bigger so I thought only SSR would lift some rendering load.

I just posted [a question on stackoverflow](https://stackoverflow.com/questions/58464758/firebase-cloud-functions-wont-work-with-react-hooks-on-nextjs-ssr) referencing this issue and a fork of your reproducible bug repo.

**Comment by Siegrift** [comment]
Yes, I am using zeit hosting (default for nextjs) and I configured firebase
manually. This was the best I could do in a few days and I didn't want to
spend much more on it

**Comment by RahmanQureshi** [comment]
The problem superficially stems from two copies of the react source code. I'm not 100% sure how it works so I won't even try to explain the root cause of the problem. 

To fix the problem, please place the app inside the functions folder and remove package.json. Unfortunately, all the dependencies of app must necessarily be dependencies of functions. Then, change next.config.js to this:

> 'use strict';
> 
> module.exports = {
>   distDir: './next'
> };
>

**Comment by jthegedus** [comment]
Hi, I wrote the initial example. This example hasn't been kept up to date as I was maintaining my own example and the example in the nextjs repo. My most up to date example does function with React hooks if you would like to check that out https://github.com/jthegedus/firebase-gcp-examples/tree/master/functions-nextjs

**Comment by jhuleatt** [comment]
We removed the `nextjs-with-firebase-hosting` sample in #804. Firebase now supports Next.js natively through [Firebase App Hosting](https://firebase.google.com/docs/app-hosting).
